### PR TITLE
Created a new contract details modal

### LINKED ISSUE #15226: Create a new contract details modal
## Resources
[Figma Design file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=910%3A1321)

## Acceptance Criteria

1. TBD
Screen Shot 2022-07-13 at 15 52 23

### PR DESCRIPTION
**Pull request by adnansahovic**
## Explanation

Created a new contract details modal. This modal can't yet be added to the app. Instead, I added a storybook that displays this modal.

## More Information

* Fixes #15226

## Screenshots/Screencaps
Screenshot 2022-08-22 at 15 22 59

## Manual Testing Steps

This modal can be tested by running storybook (yarn storybook) and then go to the COMPONENTS tab and open in App/Modals/ContractDetailsModal.

### CODE REVIEW

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    border-bottom: 1px solid var(--color-border-muted);
+
+    &__header {
+      font-style: normal;
+      font-weight: 700;
+      font-size: 16px;
+      line-height: 24px;
+      display: flex;
+      align-items: center;
+    }
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:16]: All of this should be achieveable with our <Typography> component and the boxProps on that component. When translating styles from figma we need to make sure we are translating to our design system components and standards as best as possible. For documentation see storybook https://metamask.github.io/metamask-storybook/index.html?path=/story/getting-started-introduction--page.

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      font-style: normal;
+      font-weight: 400;
+      font-size: 12px;
+      line-height: 20px;
+      display: flex;
+      align-items: center;
+      letter-spacing: 0.25px;
+      padding-top: 8px;
+      color: var(--color-text-alternative);
+    }
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:28]: Most, if not all of this, can be replaced with the Typography component. line-height/letter-spacing may have to remain cc @georgewrmarshall might wanna take a look at supporting this. Honestly though line-height is part of our typography styles that are baseline, not prop driven, and probably shoudln't be overwritten unless for a good reason. Also not sure about the letter spacing at first glance.

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    &__contract-subtitle {
+      font-style: normal;
+      font-weight: 400;
+      font-size: 14px;
+      line-height: 20px;
+      display: flex;
+      align-items: center;
+      padding-top: 16px;
+      padding-bottom: 8px;
+    }
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:39]: Same regarding Typography

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      &__address {
+        color: var(--color-text-alternative);
+        font-style: normal;
+        font-weight: 400;
+        font-size: 14px;
+        width: 176px;
+        height: 22px;
+        line-height: 22px;
+        margin-bottom: 16px;
+      }
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:69]: Mostly the same regarding typography, though i'm nervous about the number of variations of line-height you're adding as well as the hardcoded widths and heights. I'll review the implementation but I fear that these will end up making the popover brittle if were to ever resize it.

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+      &__buttons {
+        margin: 18px 20px 40px 1px;
+        display: flex;
+        justify-content: space-between;
+        width: 100%;
+        background-color: var(--color-background-default);
+      }
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:77]: Can mostly, if not entirely, be replaced by the Box component props

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+      &__buttonCopy {
+        background-color: var(--color-background-default);
+      }
+    }
+  }
+
+  &__footer {
+    @extend %row-nowrap;
```
Comment (brad-decker) [inline ui/components/app/modals/contract-details-modal/index.scss:90]: Don't extend anything defined in the ui/css/itcss folder. That is a deprecated folder that I honestly thought I had killed long ago. Use a Box, flex properties etc.

**Review by brad-decker — APPROVED** [review]
Initial review: Looking good, thank you for your submission! Please focus on reducing the amount of added css by relying on the design system components to do more of the work for you.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Meant to request changes :(

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 1 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: Hi @brad-decker, You can see changes here: [d1ae5af](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)  and [5e6a58a](https://github.com/MetaMask/metamask-extension/pull/15549/commits/5e6a58ac132773166eebd4c425cde0751a724c06)

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 2 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: @brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 3 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: @brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 4 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: @brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 5 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: @brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 6 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: @brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/d1ae5af192efb5c81d7eefe7c028e6a1223c6e37)

**Comment by mirjanaKukic** [comment]
Verified by QA

**Review by brad-decker — DISMISSED** [review]

File: ui/components/app/modals/contract-details-modal/index.scss [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        &__block-explorer.btn-link {
+          padding: 0;
+        }
+      }
+    }
+  }
+
+  &__footer {
+    button + button {
+      margin-left: 1rem;
```
Comment (darkwing) [inline ui/components/app/modals/contract-details-modal/index.scss:28]: `margin-inline-start` would be better for RTL

File: ui/components/ui/icon/copy-contract-details.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    className={className}
+    width="18"
+    height="18"
+    viewBox="0 0 18 18"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M12.3333 9.75008V13.2501C12.3333 16.1667 11.1667 17.3334 8.24999 17.3334H4.74999C1.83332 17.3334 0.666656 16.1667 0.666656 13.2501V9.75008C0.666656 6.83341 1.83332 5.66675 4.74999 5.66675H8.24999C11.1667 5.66675 12.3333 6.83341 12.3333 9.75008Z"
+      fill="#BBC0C5"
```
Comment (darkwing) [inline ui/components/ui/icon/copy-contract-details.js:15]: @georgewrmarshall Do we have a `var()` for this?

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/components/ui/icon/copy-contract-details.js (hunk 8 above)
Comment (brad-decker) [reply ui/components/ui/icon/copy-contract-details.js:15]: ooh, can we pass --vars as html attributes for these like in the case of fill?

File: ui/components/app/modals/contract-details-modal/index.scss (hunk 7 above)
Comment (adnansahovic) [reply ui/components/app/modals/contract-details-modal/index.scss]: Hi @darkwing Done [here](https://github.com/MetaMask/metamask-extension/pull/15549/commits/c7c6bd991d609b3544b8a5df4a12692ccca100b1#diff-7258ea0ef1005b16933bd9083a0d3013abc97dbc7ec498286b54a5e7633c7d75).

**Review by darkwing — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Comment by brad-decker** [comment]
CI missed the commit hook, so updated branch. will merge once green

**Pull request merged**